bucketlist/bucketlist_api.py

```diff
@@ -0,0 +1,235 @@
+                    'The bucketlist with ID %s was deleted' % list_id}, 200
+        else:
+            return {'message': 'Bucketlist not found'}, 404
+
+    def create_bucketlist_item(self, list_id):
+        """
+        Create a new item in the list specified by list_id
+        """
+        if not list_id:
+            return {'message': 'That list was not found'}, 404
+
+        bucketlist = Bucketlist.query.filter_by(
+            created_by=current_identity.user_id, list_id=list_id).first()
+
+        if bucketlist is not None:
+            parser = reqparse.RequestParser()
+            parser.add_argument('item_content', required=True,
+                                help="Item content cannot be blank")
+            args = parser.parse_args()
+
+            item = BucketlistItem(args.item_content, list_id, False)
+            db.session.add(item)
+            db.session.commit()
+
+            response = marshal(item, item_fields)
+
+            return {'item': response, 'message':
+                    'Item created in bucketlist with ID %s' % list_id}, 201
+        else:
+            return {'message': 'That list was not found'}, 404
+
+    def update_bucketlist_item(self, list_id, item_id):
+        """
+        Update a list item with id item_id
+
+        The item must be in the list specified by list_id
+        """
+        if not item_id or not list_id:
+            return {'message': 'That item was not found'}, 404
+
+        item = BucketlistItem.query.filter_by(
+            item_id=item_id, bucketlist=list_id).first()
+
+        if item is not None:
+            parser = reqparse.RequestParser()
+            parser.add_argument('item_content', required=True,
+                                help="Item content cannot be blank")
+            parser.add_argument('done')
+            args = parser.parse_args()
+
+            if args.item_content:
+                item.item_content = args.item_content
+            if args.done:
+                item.done = args.done
+
+            db.session.commit()
+
+            response = marshal(item, item_fields)
+
+            return {'item': response, 'message':
+                    'Item with ID %s was updated' % item.item_id}, 200
+        else:
+            return {'message': 'That item was not found'}, 404
+
+    def delete_bucketlist_item(self, list_id, item_id):
+        """
+        Delete a list item with id item_id
+
+        The item must be in the list specified by list_id
+        """
+        if not item_id or not list_id:
+            return {'message': 'That item was not found'}, 404
+
+        item = BucketlistItem.query.filter_by(
+            item_id=item_id, bucketlist=list_id).first()
+
+        if item is not None:
+            db.session.delete(item)
+            db.session.commit()
+            return {'message':
+                    'Item with ID %s was deleted' % item_id}, 200
+        else:
+            return {'message': 'That item was not found'}, 404
```

Comment:
Modularize the functions to separate file that holds bucketlist and another for the bucketlist item